lib/rsvp/enumerator.js

```diff
@@ -69,27 +71,25 @@ export default class Enumerator {
 
   _settleMaybeThenable(entry, i) {
     let c = this._instanceConstructor;
-    let resolve = c.resolve;
 
-    if (resolve === originalResolve) {
+    if (c.resolve === ownResolve) {
       let then = getThen(entry);
 
-      if (then === originalThen &&
-        entry._state !== PENDING) {
+      if (then === ownThen && entry._state !== PENDING) {
         entry._onError = null;
         this._settledAt(entry._state, i, entry._result);
       } else if (typeof then !== 'function') {
         this._remaining--;
         this._result[i] = this._makeResult(FULFILLED, i, entry);
-      } else if (c === Promise) {
+      } else if (this.isUsingOwnPromise) {
         let promise = new c(noop);
         handleMaybeThenable(promise, entry, then);
         this._willSettleAt(promise, i);
       } else {
-        this._willSettleAt(new c(resolve => resolve(entry)), i);
```

Comment:
if that true, it will not break anyones code since there is no public api (that i am aware) to use native Promise (or lib Promise), so when such api appears we will need to just expect support for `Promise.resolve`